db/migrate/20180910155506_add_uniqueness_of_order_id_to_spree_shipments.rb

```diff
@@ -16,8 +16,67 @@ def destroy_all_but_latest_shipments
       group(:order_id).
       having("count(*) > 1")
 
+    all_duplicated_shipments = Spree::Shipment.
+      joins("INNER JOIN (#{latest_shipments.to_sql}) latest_shipments ON spree_shipments.order_id = latest_shipments.order_id")
+    backup_to_csv(all_duplicated_shipments)
+
     Spree::Shipment.
       joins("INNER JOIN (#{latest_shipments.to_sql}) latest_shipments ON spree_shipments.order_id=latest_shipments.order_id AND spree_shipments.updated_at != latest_shipments.updated_at").
       destroy_all
   end
+
+  def backup_to_csv(shipments)
+    CSV.open(csv_path, "w") do |csv|
+      csv << csv_header_row
+
+      shipments.each do |shipment|
+        csv << shipment_csv_row(shipment)
+      end
+    end
+  end
+
+  def csv_header_row
+    %w(
+      id
+      shipment.tracking
+      number
+      cost
+      shipped_at
+      order_id
+      shipping_method_id
+      address_id
+      created_at
+      updated_at
+      state
+    )
+  end
+
+  def shipment_csv_row(shipment)
+    [
+      shipment.id,
+      shipment.tracking,
+      shipment.number,
+      shipment.cost,
+      shipment.shipped_at,
+      shipment.order_id,
+      shipment.shipping_method_id,
+      shipment.address_id,
```

Comment:
The three above would be more useful to the instance owner if these were order number, shipping method name, and even just JSON of the address object.

Reply:
:+1: done in the new commit